src/utils/dataStructureUtils.ts

```diff
@@ -0,0 +1,17 @@
+import {IKeyObject} from "../interfaces/IKeyObject";
+
+/**
+ * Sort a map of IKeyObject by key
+ * @param {<string, IKeyObject>)} map to sort
+ * @Return sorted map
+ */
+function sortMapByKey(map: Map<string, IKeyObject>) {
+  return (Array.from(map.values()) as IKeyObject[]).sort((obj1, obj2) => {
```

Comment:
Looking at IKeyObject (the interface), this can probably be simplified:

`function sortMapByKey(map: Map<string, {key: unknown}>) {`

and no need for the `as...` syntax since we know the types
`return Array.from(map.values()).sort((obj1, obj2) => {`

The reason I suggest these changes is because the code written in this PR won't provide any typesafey, since the key itself is optional and any other key can be passed, we don't benefit at all from this type.

Let's keep it minimal, and if we'll need more references to something with a key we can export it out